src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -359,5 +359,35 @@ private void LogDownloadFailureTelemetry(IExecutionContext executionContext, Exc
 
             executionContext.AddIssue(issue);
         }
+
+        private IList<AgentArtifactDefinition> FilterArtifactDefintions(IList<AgentArtifactDefinition> agentArtifactDefinitions)
+        {
+            var definitions = new List<AgentArtifactDefinition>();
+            foreach (var agentArtifactDefinition in agentArtifactDefinitions)
+            {
+                if (agentArtifactDefinition.ArtifactType != AgentArtifactType.Custom)
+                {
+                    definitions.Add(agentArtifactDefinition);
+                }
+                else
+                {
+                    string artifactType = string.Empty;
+                    var artifactDetails = JsonConvert.DeserializeObject<Dictionary<string, string>>(agentArtifactDefinition.Details);
+                    if (artifactDetails.TryGetValue("ArtifactType", out artifactType))
+                    {
+                        if (artifactType == null || artifactType.Equals("Build", StringComparison.OrdinalIgnoreCase))
+                        {
+                            definitions.Add(agentArtifactDefinition);
+                        }
+                    }
+                    else
```

Comment:
all the external artifacts which are not of type 'Build' . Currently we have only two types 'build' and 'sourceControl'.